jetty-server/src/main/java/org/eclipse/jetty/server/HttpChannel.java

```diff
@@ -406,7 +410,12 @@ public boolean handle()
                             // the following is needed as you cannot trust the response code and reason
                             // as those could have been modified after calling sendError
                             Integer code = (Integer)_request.getAttribute(RequestDispatcher.ERROR_STATUS_CODE);
-                            _response.setStatus(code != null ? code : HttpStatus.INTERNAL_SERVER_ERROR_500);
+                            if (code == null)
+                                code = HttpStatus.INTERNAL_SERVER_ERROR_500;
+                            _response.setStatus(code);
+
+                            // Add Connection:close if we can't consume the input
+                            ensureContentConsumedOrConnectionClose();
```

Comment:
I think this needs a comment on why it is done here, because otherwise it should be done only in `COMPLETE`.